src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Swagger;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class DocumentationNormalizer implements NormalizerInterface
+{
+    /** @var NormalizerInterface */
+    private $decoratedNormalizer;
+
+    /** @var RequestStack */
+    private $requestStack;
+
+    public function __construct(NormalizerInterface $decoratedNormalizer, RequestStack $requestStack)
+    {
+        $this->decoratedNormalizer = $decoratedNormalizer;
+        $this->requestStack = $requestStack;
+    }
+
+    public function supportsNormalization($data, string $format = null): bool
+    {
+        return $this->decoratedNormalizer->supportsNormalization($data, $format);
+    }
+
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        $docs = $this->decoratedNormalizer->normalize($object, $format, $context);
+
+        if ($this->isAdminDocs()) {
```

Comment:
What would you say, about creating two separate services, which will provide docs for admin and shop? Let's leave the content of this class as it is, just to extract the body of the if statement (env everything after it, but before `array_merge_recursive`. It should be easier to test and adjust the code if one would need it.